os_ver_details.sh

```diff
@@ -37,23 +29,41 @@ get_os_ver_details()
   else
       # Fall back to uname, e.g. "Linux <version>", also works for BSD, etc.
       OS=$(uname -s)
+      export OS
       VER=$(uname -r)
+      export VER
   fi
 }
 
+#
+# NOTE: We set NUM_THREADS to NUM_CORES / 4 to ensure the build process doesn't overrun
+#       the amount of memory on the build host. It's not an exact science, but it helps
+#       prevent the OOM process from killing compilers and causing issues with the
+#       build.
+#
 get_num_cores()
 {
-   nproc_exist=$(which nproc)
-   if [ ! -z "$nproc_exist" ];
+   # First check if NUM_CORES and NUM_THREADS are already set, and exit if so.
+   if [[ -n $NUM_CORES && -n $NUM_THREADS ]]
+   then
+      echo "NUM_CORES and NUM_THREADS already set."
+      return
+   fi
+
+   nproc_exist=$(command -v nproc)
+   if [ -n "$nproc_exist" ];
    then
        NUM_CORES=$(nproc --all)
        echo "Num cores on a system: $NUM_CORES"
        if [ "$NUM_CORES" -gt 4 ]
        then
-           NUM_THREADS=$(($NUM_CORES / 4))
+           NUM_THREADS=$((NUM_CORES / 4))
            NUM_THREADS=-j$NUM_THREADS
+       else
+           NUM_THREADS=-j${NUM_CORES}
        fi
-else
-    NUM_CORES=""
-fi
+    else
+        NUM_CORES=1
+        NUM_THREADS=j1
```

Comment:
-j1?

Reply:
Yes, if you see where this is used ([here](https://github.com/ipdk-io/ipdk/pull/28), for example), there is no `-j` there.

Reply:
You'll see this is what is used in the IPDK file (which this is just a copy of) [here](https://github.com/ipdk-io/ipdk/blob/main/build/scripts/os_ver_details.sh#L61). I will push another fix to correct [here](https://github.com/ipdk-io/ipdk/blob/main/build/scripts/os_ver_details.sh#L63) and [here](https://github.com/ipdk-io/ipdk/blob/main/build/scripts/os_ver_details.sh#L67) in a minute in the IPDK version of this file.